Detectors/TOF/compression/include/TOFCompression/CompressorTask.h

```diff
@@ -33,14 +33,15 @@ template <typename RDH, bool verbose, bool paranoid>
 class CompressorTask : public Task
 {
  public:
-  CompressorTask() = default;
+  CompressorTask(long payloadLim = -1) : mPayloadLimit(payloadLim) {}
   ~CompressorTask() override = default;
   void init(InitContext& ic) final;
   void run(ProcessingContext& pc) final;
 
  private:
   Compressor<RDH, verbose, paranoid> mCompressor;
   int mOutputBufferSize;
+  long mPayloadLimit = -1;
```

Comment:
For the record, the payload size is uint64_t, so this might actually have some race condition if the payload is larger than 2GB for real. I assume that will never be the case, though.

Reply:
This is the limit we pass as external param, I don't see issue if this limit doesn't exceed that boundary (it is set manually).
-1 is used as a switch to apply or not apply the check